app/validators/uri_validator.rb

```diff
@@ -26,6 +26,10 @@ def valid_scheme?(scheme)
     end
   end
 
+  def valid_host?(host)
+    host.present? && (host.size <= 255) && !host.split('.').any? { |label| label.size > 63 }
```

Comment:
I would separate it in another method
```suggestion
    host.present? && valid_parts?
```

```ruby
def valid_domain_parts
   host.split('.').map(&:size).any?(&63.method(:'>'))
end